### DISCUSSION TITLE
Shoehorning line chart into ribbon chart

### DISCUSSION THREAD

**Discussion by zell-mbc** [patrykandpatrick/vico#893] [Questions] [answered]
### Question

I have the need to offer a ribbon chart and tried to implement this with an upper and lower line. 
The best I could come up with so far is upper line with default area fill and lower line with area fill set to chart background color (black or white). 

This does the job until someone turns on the chart grid. The grid gets blanked out below the lower line (#1) because I am drawing all white (alpha=FF). I played with different alpha values but all I achieved is that the top line color starts to shine through.
Given that the fill for the top line respects the grid (#2) I was wondering if there's a way to do this for the bottom part too?

![image](https://github.com/user-attachments/assets/9890507a-1a2a-416e-80ed-2a67a4053ad5)

### Vico version(s)

v2.beta.1

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment, marked as answer]
Hello, @zell-mbc.
> Given that the fill for the top line respects the grid

The fill is translucent, so the grid shines through. The fill doesn't know anything about the grid.

> I was wondering if there's a way to do this for the bottom part too?

For the grid to be fully visible below the ribbon, you'd need it not to be covered up by any fill. You can't erase part of one fill with another.

Thus, you'd need to have only one fill, which would be between the two lines. `LineCartesianLayer` doesn't support this out of the box, since it's intended for line charts, but you can achieve this through subclassing.

We plan on adding ribbon charts in the future. See #460.

**Comment by Tyler-Lopez** [comment]
Hi there @zell-mbc - I'd like to also share my implementation for the functionality you are looking at in a related question: https://github.com/patrykandpatrick/vico/discussions/855#discussioncomment-10847038 

If you find anything much better in your explorations, I'd love to know!

**Reply by zell-mbc** [reply]
I will definitely check this out! Thanks for letting me know :+1:
